### PR TITLE
convert units to mm in the js scripts. closes #14.

### LINKED ISSUE #14: Configurator doesn't show target locations correctly if the target units are in inches
My instance settings seem to change the units of the targets to inches which seems to also break the visualization.
Screenshot 2024-11-22 at 9 35 14 AM
Screenshot 2024-11-22 at 9 37 15 AM

Changing the units resolves the issue, but I imagine it's confusing for certain US based users.

### PR DESCRIPTION
**Pull request by akarras**
I haven't figured out how to test this- I'm fairly confident this will work, but haven't been able to verify. 😓 

didn't notice that zed formatted the code until I went to open the PR. I can revert if desired, but I will be lazy and not fix it unless you ask 😄

### CODE REVIEW

File: everything-presence-mmwave-configurator/www/script.js [hunk 1]
```diff
@@ -72,7 +72,36 @@ document.addEventListener("DOMContentLoaded", () => {
   function calculateOffsetY() {
     let absAngle = Math.abs(installationAngle);
     if (absAngle <= 30) offsetY = 0;
-    else offsetY = detectionRange * Math.sin((absAngle - 30) * Math.PI / 180);
+    else offsetY = detectionRange * Math.sin(((absAngle - 30) * Math.PI) / 180);
+  }
+
+  /// Returns the entity's state converted from whatever unit is configured in the UI converted to millimeters
+  function getEntityStateMM(entity) {
```
Comment (MenesesPT) [inline everything-presence-mmwave-configurator/www/script.js:79]: We have to round the values that this function returns. I've adapted the function below:

```JS
/// Returns the entity's state converted from whatever unit is configured in the UI converted to millimeters
function getEntityStateMM(entity) {
  const state = entity ? parseFloat(entity.state) || 0 : 0;
  let result = state;
  // cm, in, ft, km, m, mi, nmi, yd and mm are supported in home assistant.
  switch (entity.attributes.unit_of_measurement) {
    case 'mm':
      break; // Avoid checking every unit for the most common case
    case "in":
      result = state * 25.4; // Convert inches to millimeters
      break;
    case "ft":
      result = state * 304.8; // Convert feet to millimeters
      break;
    case "km":
      result = state * 1000000; // Convert kilometers to millimeters
      break;
    case "in":
      result = state * 1000; // Convert meters to millimeters
      break;
    case "in":
      result = state * 1.609e6; // Convert miles to millimeters
      break;
    case "in":
      result = state * 1.852e6; // Convert nautical miles to millimeters
      break;
    case "in":
      result = state * 914.4; // Convert yards to millimeters
      break;
  }
  return Math.round(result);
}
```

**Review by MenesesPT — CHANGES_REQUESTED** [review]
I've tested this and seems to work for the most part. However on the conversion sometimes values need to be truncated (i.e. 12 inches would result in 304.79999999999995 mm). As this values are shown on the table it gets really hard to read since the width of the cells keep changing.
I've suggested a potential fix.

File: everything-presence-mmwave-configurator/www/script.js (hunk 1 above)
Comment (akarras) [reply everything-presence-mmwave-configurator/www/script.js:79]: Thanks for testing!
I Integrated your suggestion- I did fixup the last couple cases for miles/nautical miles/yards that were still case 'in' . This looks like it's working well on my end though.

Comment (MenesesPT) [reply everything-presence-mmwave-configurator/www/script.js:79]: Ups 😅 thanks!

**Review by MenesesPT — APPROVED** [review]

**Comment by EverythingSmartHome** [comment]
Thanks guys, works great from my testing!

**Pull request merged**